pkg/storage/dirtytracker/dirty_tracker.go

```diff
@@ -39,14 +39,24 @@ type Metrics struct {
 }
 
 func (dtr *Remote) GetMetrics() *Metrics {
-	minAge := dtr.MeasureDirtyAge()
+	// Figure out oldest dirty block...
+	dtr.dt.trackingLock.Lock()
+	maxAgeDirty := time.Duration(0)
+	minAge := time.Now()
+	for _, t := range dtr.dt.trackingTimes {
+		if t.Before(minAge) {
+			minAge = t
+			maxAgeDirty = time.Since(minAge)
+		}
+	}
+	dtr.dt.trackingLock.Unlock()
```

Comment:
I was expecting a `defer` here, and while `time.Since` is fairly low overhead, this reads clearer to me:

```golang
dtr.dt.trackingLock.Lock()
defer dtr.dt.trackingLock.Unlock()

// Find the oldest tracking time.
minTime := time.Now()
hasDirtyBlocks := false
for _, t := range dtr.dt.trackingTimes {
	if t.Before(minTime) {
		minTime = t
		hasDirtyBlocks = true
	}
}

var maxAgeDirty time.Duration
if hasDirtyBlocks {
	maxAgeDirty = time.Since(minTime)
}
```

WDYT @jimmyaxod